arangod/Cache/BucketState.h

```diff
@@ -47,25 +46,20 @@ struct BucketState {
   //////////////////////////////////////////////////////////////////////////////
   /// @brief Flags which can be queried or toggled to reflect state.
   ///
-  /// Each flag must have exactly one bit set, fit in uint32_t. The 'locked'
+  /// Each flag must have exactly one bit set, fit in uint16_t. The 'locked'
   /// flag is special and should remain the least-significant bit. When other
   /// flags are added,they should be kept in alphabetical order for readability,
   /// and all flag values should be adjusted to keep bit-significance in
   /// ascending order.
   //////////////////////////////////////////////////////////////////////////////
-  enum class Flag : std::uint32_t {
```

Comment:
note: the removed enum values were not in use in the existing code, so they are removed in this PR.
the change from uint32_t to uint16_t was necessary so we are able to additionally squeeze a uint16_t `_slotsUsed` counter into the 128 bytes that a PlainBucket or TransactionalBucket occupies, without making the struct any bigger.